[19.0][MIG] website_sale_product_pack: Migration to 19.0

lef-adhoc opened the pull request:
Depends on:

- [ ] <https://github.com/OCA/product-pack/pull/244>

chienandalu committed:
[ADD] website_sale_product_pack: New module

Compatibility module between sale_product_pack and website_sale

TT30385

chienandalu committed:
[ADD] website_sale_product_pack: New module

Compatibility module between sale_product_pack and website_sale

TT30385

chienandalu committed:
[FIX] stock_product_pack: get already computed values if existing

Done for e-commerce compatibility purposes, althoug it's more performant
indeed.

chienandalu committed:
[FIX] stock_product_pack: get already computed values if existing

Done for e-commerce compatibility purposes, althoug it's more performant
indeed.

oca-travis committed:
[UPD] Update website_sale_product_pack.pot

oca-travis committed:
[UPD] Update website_sale_product_pack.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX] website_sale_product_pack: Resilient tests

Create and select a specific pricelist for avoiding problems in
integrated environments where the default pricelist currency has been
changed.

pedrobaeza committed:
[FIX] website_sale_product_pack: Resilient tests

Create and select a specific pricelist for avoiding problems in
integrated environments where the default pricelist currency has been
changed.

chienandalu committed:
[FIX] website_sale_product_pack: detailed totalized packs

When the cart is confirmed, a price recalculation is triggered for every
order line. This is wrong for detailed totalized packs, which lines
should be at 0.

TT38186

chienandalu committed:
[FIX] website_sale_product_pack: detailed totalized packs

When the cart is confirmed, a price recalculation is triggered for every
order line. This is wrong for detailed totalized packs, which lines
should be at 0.

TT38186

chienandalu committed:
Set portal user pricelist

chienandalu committed:
Set portal user pricelist

OCA-git-bot committed:
website_sale_product_pack 13.0.1.0.1

OCA-git-bot committed:
website_sale_product_pack 13.0.1.0.1

OCA-git-bot committed:
[IMP] website_sale_product_pack: Published and unpublished for products using packs

OCA-git-bot committed:
[IMP] website_sale_product_pack: Published and unpublished for products using packs

OCA-git-bot committed:
website_sale_product_pack 13.0.1.0.2

OCA-git-bot committed:
website_sale_product_pack 13.0.1.0.2

Commit:
[UPD] Update website_sale_product_pack.pot

Commit:
[UPD] Update website_sale_product_pack.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

alvaro-domatix committed:
[16.0][MIG] website_sale_product_pack: Migration to 16.0

alvaro-domatix committed:
[16.0][MIG] website_sale_product_pack: Migration to 16.0

augusto-weiss committed:
[FIX] website_product_pack: detailed displayed components price on website

augusto-weiss committed:
[FIX] website_product_pack: detailed displayed components price on website

lef-adhoc committed:
[FIX] website_sale_product_pack: Fixed KeyError: 'product_id' error due to Odoo making changes. https://github.com/odoo/odoo/commit/b121f8e1d3ca5b98e71e25710427e6210afe9ef6

lef-adhoc committed:
[FIX] website_sale_product_pack: Fixed KeyError: 'product_id' error due to Odoo making changes. https://github.com/odoo/odoo/commit/b121f8e1d3ca5b98e71e25710427e6210afe9ef6

Commit:
[UPD] Update website_sale_product_pack.pot

Commit:
[UPD] Update website_sale_product_pack.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-pack-16.0/product-pack-16.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-16-0/product-pack-16-0-website_sale_product_pack/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-pack-16.0/product-pack-16.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-16-0/product-pack-16-0-website_sale_product_pack/it/

augusto-weiss committed:
[IMP] website_sale_product_pack: pre-commit execution

augusto-weiss committed:
[IMP] website_sale_product_pack: pre-commit execution

augusto-weiss committed:
[MIG] website_sale_product_pack: Migration to 17.0

augusto-weiss committed:
[MIG] website_sale_product_pack: Migration to 17.0

Commit:
[UPD] Update website_sale_product_pack.pot

Commit:
[UPD] Update website_sale_product_pack.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-pack-17.0/product-pack-17.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-17-0/product-pack-17-0-website_sale_product_pack/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-pack-17.0/product-pack-17.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-17-0/product-pack-17-0-website_sale_product_pack/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-pack-17.0/product-pack-17.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-17-0/product-pack-17-0-website_sale_product_pack/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-pack-17.0/product-pack-17.0-website_sale_product_pack
Translate-URL: https://translation.odoo-community.org/projects/product-pack-17-0/product-pack-17-0-website_sale_product_pack/it/

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

Abranes committed:
[17.0][FIX] website_sale_product_pack: add decorator to remove warnings

Abranes committed:
[17.0][FIX] website_sale_product_pack: add decorator to remove warnings

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

ALopez-Adhoc committed:
[IMP] website_sale_product_pack: pre-commit execution

ALopez-Adhoc committed:
[IMP] website_sale_product_pack: pre-commit execution

ALopez-Adhoc committed:
[MIG] website_sale_product_pack: Migration to 18.0

ALopez-Adhoc committed:
[MIG] website_sale_product_pack: Migration to 18.0

lef-adhoc committed:
[IMP] website_sale_product_pack: pre-commit execution

lef-adhoc committed:
[IMP] website_sale_product_pack: pre-commit execution

lef-adhoc committed:
[MIG] website_sale_product_pack: Migration to 19.0

lef-adhoc committed:
[MIG] website_sale_product_pack: Migration to 19.0

nicolascol reviewed (approved):
LGTM Functional review

much-filipe-colaco commented:
Any news on this migration? Is it planned to be released soon?

lef-adhoc committed:
[FIX] website_sale_product_pack: keep discount gate on strikethrough price

The cart_lines_price override replaced the whole t-if of the <del>
element with 'not line.pack_parent_line_id', dropping the core
discount gate that v19 moved into that same attribute. This forced
the strikethrough price on every non-component cart line, even
without a discount, showing a struck price equal to the net price.

Keep the core gate (_should_show_strikethrough_price) and only add
the pack-component exclusion on top of it.
